### PR TITLE
Add choco package files and readme

### PR DESCRIPTION
**Pull request by jneira**
* It adds the needed files to setup the choco package, taking inspiration from GhcChoco and CabalChoco
* Not sure if the powershell code will be too advanced for some windows versions

@Mistuke could you take a look if you get a chance? many thanks in advance

### CODE REVIEW

**Comment by Mistuke** [comment]
> * It adds the needed files to setup the choco package, taking inspiration from GhcChoco and CabalChoco
> * Not sure if the powershell code will be too advanced for some windows versions

The chocolatey verifier service will check this on upload. It always checks against the least common denominator. 

> 
> @Mistuke could you take a look if you get a chance? many thanks in advance

Awesome! This looks good to me, the only comment I have is that I've recently moved to using the tools install location https://github.com/Mistuke/GhcChoco/commit/93676e55aca7629de5b244393117b9decb503ba1 which is a bit faster as it doesn't force the hashing requirements of any files you download and also removes the need for admin rights and allows people to change it by setting environment variables.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,47 @@
...
+
+Chocolatey sources for [haskell-language-server](https://github.com/haskell/haskell-language-server/blob/master/README.md) installs.
+
+This repository contains the sources for the haskell-language-server Chocolatey packages.
+
+## Usage
+
+To use these get [Chocolatey](https://chocolatey.org/) and then:
+
+and then:
```
Comment (jneira) [inline README.md]:
```suggestion
```

remove superflous line

File: 1.1.0/haskell-language-server/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+$ghcVersions = @()
+
+if ($pp['for-all-ghcs']) {
+  $ghcVersions = $supportedGhcVersions
+  Write-Host "Installing $packageName for all supported ghc versions: $($ghcVersions -join ', ')"
+} elseif ($pp['for-ghcs']) {
+  $ppForGhcs = $pp['install-for-ghc-versions']
+  $ppForGhcs = if (-not $ppForGhcs) { $pp['for-ghcs'] } else { $ppForGhcs }
```
Comment (jneira) [inline 1.1.0/haskell-language-server/tools/chocolateyinstall.ps1]:
```suggestion
  $ppForGhcs = $pp['for-ghcs']
```

the alias is not longer used

File: 1.1.0/haskell-language-server/tools/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  exit -1
+}
+
+$supportedInstalledGhcs = (Compare-Object $ghcVersions $supportedGhcVersions `
+                            -IncludeEqual -ExcludeDifferent `
+                          ).InputObject
+
+if ($supportedInstalledGhcs.Count -le 0) {
+  Write-Host `
+    "None of the selected ghc versions is supported by $packageName so I can't determine which binaries should be downloaded."
```
Comment (jneira) [inline 1.1.0/haskell-language-server/tools/chocolateyinstall.ps1:92]:
```suggestion
  Write-Host `
    "None of the selected ghc versions is supported by $packageName so I can't determine which binaries should be downloaded."
  Write-Host "The supported ghc versions are $($supportedGhcVersions -join ', ')"
```

print the supported ghc version to help user recover the error

File: 1.1.0/haskell-language-server/tools/chocolateyinstall.ps1 [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+if ($pp['for-all-ghcs']) {
+  $ghcVersions = $supportedGhcVersions
+  Write-Host "Installing $packageName for all supported ghc versions: $($ghcVersions -join ', ')"
+} elseif ($pp['for-ghcs']) {
+  $ppForGhcs = $pp['install-for-ghc-versions']
+  $ppForGhcs = if (-not $ppForGhcs) { $pp['for-ghcs'] } else { $ppForGhcs }
+  $forGhcs = $ppForGhcs.split("|")
+  $ghcVersions = $supportedGhcVersions.Where({$forGhcs.Contains($_)})
+  if ($ghcVersions.Count -le 0) {
+    Write-Host "It as not been possible select any supported ghc versions using the parameter value $ppForGhcs"
```
Comment (jneira) [inline 1.1.0/haskell-language-server/tools/chocolateyinstall.ps1]:
```suggestion
    Write-Host "It as not been possible to select any supported ghc versions using the parameter value $ppForGhcs"
```

**Comment by jneira** [comment]
@Mistuke sorry to bother you again, i changed the package to use `Get-ToolsLocation` which can be controlled with the env var `ChocolateyToolsLocation` (which is good)
But that path is not in the global path by default (at least with my choco installation) and executables are not getting shimmed anymore, so how is suppose to have them in PATH, changing it manually?

**Comment by Mistuke** [comment]
> @Mistuke sorry to bother you again, i changed the package to use `Get-ToolsLocation` which can be controlled with the env var `ChocolateyToolsLocation` (which is good)
> But that path is not in the global path by default (at least with my choco installation) and executables are not getting shimmed anymore, so how is suppose to have themin PATH, changing it manually?

It's no bother at all! 

Indeed, the automatic shimgen is only done for files in the package folder. Instead of opt out, it's now opt-in. You can easily tell it to shim them using `Install-BinFile`, note that you'll need `UnInstall-BinFile` as well in your cleanup script. The advantage of this is that it gives you flexibility in how to handle side-by-side installs.

**Pull request merged**